lib/Mojo/Promise.pm

```diff
@@ -348,6 +355,14 @@ L<Mojo::Promise> object resolving to the return value of the called handler.
     }
   );
 
+=head2 timeout
+
+  my $new = Mojo::Promise->timeout(5 => 'Timeout!');
+  my $new = $promise->timeout(5 => 'Timeout!');
```

Comment:
Should the default rejection be documented? If so, how is the best way to do that?

Reply:
At the very least a `my $new = $promise->timeout(5);` here.